substrate/frame/revive/src/wasm/mod.rs

```diff
@@ -232,6 +232,38 @@ impl<T: Config> CodeInfo<T> {
 	pub fn code_len(&self) -> u64 {
 		self.code_len.into()
 	}
+
+	/// Returns the number of times the specified contract exists on the call stack. Delegated calls
+	/// Increment the reference count of a of a stored code by one.
+	///
+	/// # Errors
+	///
+	/// [`Error::CodeNotFound`] is returned if no stored code found having the specified
+	/// `code_hash`.
+	pub fn increment_refcount(code_hash: H256) -> DispatchResult {
+		<CodeInfoOf<T>>::mutate(code_hash, |existing| -> Result<(), DispatchError> {
+			if let Some(info) = existing {
+				*info.refcount_mut() = info.refcount().saturating_add(1);
```

Comment:
Should this error out on overflow instead? Although it's likely very expensive to saturate.

Reply:
We can error out. This function used to be called from an infallible code path. So I made the refcount `u64` and called it a day.